feat(endpoint): add functionality for re-boostrapping to the network using updated contacts

lionel-faber opened the pull request.

lionel-faber committed:
feat(endpoint): add functionality for re-boostrapping to the network
using updated contacts

lionel-faber committed:
feat(boostrap_cache): add API to replace contents of bootstrap cache

bochaco commented on src/api.rs:
```diff
@@ -97,15 +97,11 @@ impl QuicP2p {
 
         let mut bootstrap_cache =
             BootstrapCache::new(cfg.hard_coded_contacts, custom_dirs.as_ref())?;
-        if use_bootstrap_cache {
```
I think it'll be useful o have some log entry stating that bootstrap cache is being used and which are the nodes being used from the cache.

lionel-faber committed:
feat(api): add public API to get an endpoint's list of bootstrap nodes

- this commit also makes some improvements to information in logs
- also adds a force argument for BootstrapCache::try_sync_to_disk

Yoga07 commented on src/tests/common.rs:
```diff
@@ -423,6 +423,7 @@ async fn many_messages() -> Result<()> {
 // bootstrap contacts later.
 #[tokio::test]
 async fn connection_attempts_to_bootstrap_contacts_should_succeed() -> Result<()> {
+    utils::init_logging();
```
Should this be done for other tests too or should be removed?

lionel1704 replied:
It is already done in most tests. This particular one was failing so I added it in for debugging. It can be added in when needed.
